src/plugins/intel_cpu/src/emitters/tpp/x64/jit_eltwise_emitters.hpp

```diff
@@ -0,0 +1,120 @@
+// Copyright (C) 2020-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <openvino/core/rt_info.hpp>
+
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/lowered/expression.hpp"
+
+#include "emitters/plugin/x64/jit_emitter.hpp"
+#include "emitters/plugin/x64/jit_load_store_emitters.hpp"
+
+#include "jit_tpp_emitter.hpp"
+
+#include <cxxabi.h>
+namespace ov {
+namespace intel_cpu {
+
+class BinaryEltwiseTppEmitter : public TppEmitter {
+public:
+    BinaryEltwiseTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                            dnnl::impl::cpu::x64::cpu_isa_t isa,
+                            const ov::snippets::lowered::ExpressionPtr& expr);
+    size_t get_inputs_num() const override { return 2; }
+    static void execute_kernel(libxsmm_meltwfunction_binary eltwise_kernel, void *in0, void *in1, void *out0);
+    const uintptr_t get_execute_function_ptr() const override { return reinterpret_cast<const uintptr_t>(execute_kernel); }
+    const uintptr_t get_compiled_kernel_ptr() const override;
+    static std::set<std::vector<element::Type>> get_supported_precisions(const std::shared_ptr<ov::Node>& node = nullptr);
+
+protected:
+    libxsmm_meltw_binary_shape m_shape;
+    libxsmm_meltw_binary_type m_op_type;
+    void validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const override;
+    static  libxsmm_blasint get_broadcasted_dim(libxsmm_blasint dim0, libxsmm_blasint dim1, std::pair<bool, bool>& bcast_flags);
+};
+//#define DEBUG_TPP_EMITTERS
+class UnaryEltwiseTppEmitter : public TppEmitter {
+public:
+    UnaryEltwiseTppEmitter(dnnl::impl::cpu::x64::jit_generator* h,
+                            dnnl::impl::cpu::x64::cpu_isa_t isa,
+                            const ov::snippets::lowered::ExpressionPtr& expr);
+    size_t get_inputs_num() const override { return 1; }
+
+    static void execute_kernel(libxsmm_meltwfunction_unary eltwise_kernel, void *in0, void *out0);
+    const uintptr_t get_compiled_kernel_ptr() const override {
+        return reinterpret_cast<const uintptr_t>(libxsmm_dispatch_meltw_unary_v2(m_op_type,
+                                                                                 m_shape,
+                                                                                 m_compile_flags));
+    }
+    const uintptr_t get_execute_function_ptr() const override { return reinterpret_cast<const uintptr_t>(execute_kernel); }
+    static std::set<std::vector<element::Type>> get_supported_precisions(const std::shared_ptr<ov::Node>& node = nullptr);
+
+protected:
+    libxsmm_meltw_unary_shape m_shape;
+    libxsmm_meltw_unary_type m_op_type;
+    void validate_arguments(const std::vector<size_t> &in, const std::vector<size_t> &out) const override;
+};
+
+class ReduceTppEmitter : public UnaryEltwiseTppEmitter {
```

Comment:
If take a look at the implementations, I understand that there are the same impls. But maybe we should rename `UnaryEltwiseTppEmitter`? At least it's not Eltwise, I think 😃 Maybe just `UnaryTppEmitter`? What do you think?
Because I'm really confused when I saw that `Reduce` emitter class is derived from `Eltwise` emitter class

Reply:
Now about renaming: if we rename `UnaryEltwiseTppEmitter` to `UnaryTppEmitter`, then we'll have to rename `BinaryEltwiseTppEmitter` to `BinaryTppEmitter` to align. However, this would clearly be inconsistent, since there are other binary operations (e.g. gemm) that are not eltwise.

Reply:
@IvanNovoselov thanks for the explanation. Taking it into account, I'd rather to leave class names as is as the least confused solution :)